### PR TITLE
feat: Github Action to compare changes in openapi-derefed.json

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
If someone makes a change to the api docs schema/paths, they should be able to compare the current derefed.json looks against their changes. 

## UI
Screen Shot 2020-10-06 at 4 37 05 PM

### CODE REVIEW

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -375,3 +375,57 @@ jobs:
             github-token: ${{ secrets.GITHUB_TOKEN }}
             gcs-bucket: 'sentry-visual-snapshots'
             gcp-service-account-key: ${{ secrets.SNAPSHOT_GOOGLE_SERVICE_ACCOUNT_KEY }}
+
+    openapi-diff:
```
Comment (billyvg) [inline .github/workflows/acceptance.yml:379]: Let's move this to a new workflow file named `openapi-diff.yml` and job as `diff` or `check`

File: .github/workflows/acceptance.yml [hunk 2]
```diff
@@ -375,3 +375,57 @@ jobs:
...
+            path: sentry-api-schema
+            token: ${{ steps.getsentry.outputs.token }}
+
+        - name: Install/setup node
+          if: steps.filter.outputs.api-docs == 'true'
+          uses: volta-cli/action@v1
+
+        - name: Install/setup json-diff
+          if: steps.filter.outputs.api-docs == 'true'
+          run: volta install json-diff
```
Comment (billyvg) [inline .github/workflows/acceptance.yml:419]: Let's remove this and instead, use `npx` to invoke it: `npx json-diff`

File: .github/workflows/acceptance.yml [hunk 3]
```diff
@@ -375,3 +375,57 @@ jobs:
...
+      if: ${{ github.ref != 'refs/heads/master' }}
+      runs-on: ubuntu-latest
+
+      steps:
+        - name: Checkout getsentry/sentry
+          uses: actions/checkout@v2
+          with:
+            path: sentry
+
+        - uses: dorny/paths-filter@v2
```
Comment (billyvg) [inline .github/workflows/acceptance.yml:389]: Let's fork this repo as tags/branches are mutable (alternatively, use a SHA)

**Review by billyvg — CHANGES_REQUESTED** [review]

File: .github/workflows/acceptance.yml (hunk 3 above)
Comment (NisanthanNanthakumar) [reply .github/workflows/acceptance.yml:389]: @billyvg if the preference is to create a new file, then we can customize the `on` trigger. So we won't need this action.

Comment (billyvg) [reply .github/workflows/acceptance.yml]: @NisanthanNanthakumar Will you want to block PR's from merging if there is a diff? If so, you'll want to keep this action as we will need to run it for all PRs. It's due to how status checks work - if you make them required then it has to be present for ALL PRs.

File: .github/workflows/openapi-diff.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    paths:
+      - 'api-docs/**'
+  pull_request:
+    paths:
+      - 'api-docs/**'
+
+jobs:
+    diff:
+      if: ${{ github.ref != 'refs/heads/master' }}
+      runs-on: ubuntu-latest
```
Comment (billyvg) [inline .github/workflows/openapi-diff.yml:15]: Can we use `ubuntu-16.04` instead of latest?

File: .github/workflows/openapi-diff.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      - master
+    paths:
+      - 'api-docs/**'
+  pull_request:
+    paths:
+      - 'api-docs/**'
+
+jobs:
+    diff:
+      if: ${{ github.ref != 'refs/heads/master' }}
```
Comment (billyvg) [inline .github/workflows/openapi-diff.yml:14]: Is this intentional? It is at odds with the `on.push` to `master`

File: .github/workflows/openapi-diff.yml (hunk 4 above)
Comment (NisanthanNanthakumar) [reply .github/workflows/openapi-diff.yml:15]: @billyvg I don't mind, but is there a reason? 16.04 stops getting updates April 2021, so wouldn't it be a future security risk?

File: .github/workflows/acceptance.yml (hunk 3 above)
Comment (NisanthanNanthakumar) [reply .github/workflows/acceptance.yml]: @billyvg nah for now it's not required. It's more for reminding that changes were made and to confirm that these were the right changes

File: .github/workflows/openapi-diff.yml (hunk 4 above)
Comment (billyvg) [reply .github/workflows/openapi-diff.yml]: Prefer explicit versions than having to guess what `latest` means

File: .github/workflows/openapi-diff.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+name: openapi-diff
+on:
+  push:
+    branches:
+      - master
```
Comment (joshuarli) [inline .github/workflows/openapi-diff.yml:5]: @billyvg Do you know if this is like on.push.branches AND on.push.paths, or OR? 

Also in either case, does this action really need to run on push @NisanthanNanthakumar?

Comment (NisanthanNanthakumar) [reply .github/workflows/openapi-diff.yml:5]: @joshuarli The consensus, for now, is that we should run on push to master to check that nothing unexpected happened on master, similar to `acceptance.yml`

File: .github/workflows/openapi-diff.yml (hunk 4 above)
Comment (NisanthanNanthakumar) [reply .github/workflows/openapi-diff.yml]: converted to `ubuntu-18.04`

File: .github/workflows/openapi-diff.yml (hunk 6 above)
Comment (joshuarli) [reply .github/workflows/openapi-diff.yml:5]: Okay.

Comment (billyvg) [reply .github/workflows/openapi-diff.yml:5]: @joshuarli actually not sure, I assumed it was AND but I don't know off hand

**Review by billyvg — APPROVED** [review]

File: .github/workflows/acceptance.yml (hunk 3 above)
Comment (billyvg) [reply .github/workflows/acceptance.yml]: If this is the case we prob don't need to run it on master pushes

Comment (NisanthanNanthakumar) [reply .github/workflows/acceptance.yml]: 👍 I'll remove that trigger

**Pull request merged**